src/Compilers/CSharp/Portable/Symbols/Source/SourceParameterSymbol.cs

```diff
@@ -125,24 +116,10 @@ internal SourceParameterSymbol WithCustomModifiersAndParamsCore(TypeSymbol newTy
 
             TypeWithAnnotations newTypeWithModifiers = this.TypeWithAnnotations.WithTypeAndModifiers(newType, newCustomModifiers);
 
```

Comment:
confused by the deleted if-statement here. was this just a bug?

Reply:
`SourceComplexParameterSymbol` is now abstract. There is now only one possible derived class in this context: `SourceComplexParameterSymbolWithCustomModifiers`.